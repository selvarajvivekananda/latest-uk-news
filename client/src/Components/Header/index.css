.span{
    color: aliceblue;
    font-size: xx-large;
    font-weight: bold;
}