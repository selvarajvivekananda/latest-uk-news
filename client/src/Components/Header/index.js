import React from "react";
import { Link } from "react-router-dom";
import Navbar from "react-bootstrap/Navbar";
import Nav from "react-bootstrap/Nav";
import { Button } from 'react-bootstrap'
import './index.css'

const Header = () => (
    <>
        <span className="span">UK News</span>
        <div style={{ background: "#a00606" }}>
            <Navbar id="nav" expand="sm" bg="dark" variant="dark">
                <Nav>
                    <Link to="/LandingPage">
                        <Button variant="outlined">Latest News</Button>
                    </Link>
                    <br></br>
                    <Link to="/Search">
                        <Button variant="outline-light">Search</Button>
                    </Link>
                </Nav>
            </Navbar>
        </div>
    </>

);

export default Header;