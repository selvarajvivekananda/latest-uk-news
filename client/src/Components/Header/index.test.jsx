import Header from './index';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe('Header', () => {
    it('Header renders correctly', () => {
        const { container, getByTestId } = render(<BrowserRouter><Header /></BrowserRouter>);
        const span_element = getByTestId('span');
        const value = span_element.innerHTML;
        expect(value).toBe('UK News');
    });
    it('Header contains link', () => {
        render(
            <BrowserRouter>
                <Header />
            </BrowserRouter>
        );
        expect(screen.getAllByRole('link')[0]).toHaveAttribute('href', '/LandingPage');
    });
    it('Header contains link', () => {
        render(
            <BrowserRouter>
                <Header />
            </BrowserRouter>
        );
        expect(screen.getAllByRole('link')[1]).toHaveAttribute('href', '/Search');
    });
    it('Header Snapshot test', () => {
        const tree = renderer.create(<BrowserRouter>
            <Header />
        </BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    });
})
