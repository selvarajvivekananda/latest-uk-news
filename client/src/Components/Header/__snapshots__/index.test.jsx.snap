// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header Snapshot test 1`] = `
Array [
  <span
    className="span"
    data-testid="span"
  >
    UK News
  </span>,
  <div
    style={
      Object {
        "background": "#a00606",
      }
    }
  >
    <nav
      className="navbar navbar-expand-sm navbar-dark bg-dark"
      id="nav"
    >
      <div
        className="navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          href="/LandingPage"
          onClick={[Function]}
        >
          <button
            className="btn btn-outlined"
            disabled={false}
            type="button"
          >
            Latest News
          </button>
        </a>
        <br />
        <a
          href="/Search"
          onClick={[Function]}
        >
          <button
            className="btn btn-outline-light"
            disabled={false}
            type="button"
          >
            Search
          </button>
        </a>
      </div>
    </nav>
  </div>,
]
`;
